code/modules/fissionreactor/objects_external.dm

```diff
@@ -0,0 +1,736 @@
+/*
+IN THIS FILE:
+objects that make up the exterior (shell) of the reactor.
+included:
+	reactor casing
+	coolant port
+	controller computer
+*/
+
+
+
+/obj/machinery/atmospherics/unary/fissionreactor_coolantport
+	name="fission reactor coolant port"
+	icon='icons/obj/fissionreactor/reactorcase.dmi'
+	icon_state="coolantcase"
+	density =1
+	anchored =1
+	var/datum/fission_reactor_holder/associated_reactor=null
+	use_power = MACHINE_POWER_USE_NONE
+	idle_power_usage = 0
+	active_power_usage = 0
+	var/pipename="fission reactor coolant port"
+	//this is so that we can call a proc with ourselves that will use a proc that shouldn't belong to this. this is probably very fragile, but just don't touch it and it'll be fine, i swear
+/obj/machinery/atmospherics/unary/fissionreactor_coolantport/proc/get_pipe_dir() //the atmos gods demand a sacrifice.
+	return dir
+
+/obj/machinery/atmospherics/unary/fissionreactor_coolantport/examine()
+	..()
+	if(associated_reactor && associated_reactor.considered_on())
+		to_chat(usr,"the outer plating looks like it could be cut,<span class='danger'> but it seems like a <u>really</u> bad idea.</span>")
+	else
+		to_chat(usr,"the outer plating looks like it could be cut.")
+
+
+/obj/machinery/atmospherics/unary/fissionreactor_coolantport/attackby(var/obj/I,var/mob/user)
+	if(iswelder(I))
+		if(associated_reactor && associated_reactor.considered_on())
+			if(user.a_intent==I_HELP)
+				to_chat(usr,"<span class='danger'>this seems like a really bad idea.</span>")
+				return
+			var/obj/item/tool/weldingtool/WT = I
+			user.visible_message("<span class='notice'>[user] starts welding \the [src]'s external plating off its frame.</span>", "<span class='notice'>You start welding \the [src]'s external plating off its frame.</span>")
+			if(WT.do_weld(user,src,60,0))
+				qdel(src)
+				var/obj/structure/girder/reactor/newcase= new /obj/structure/girder/reactor
+				newcase.loc=src.loc
+				newcase.pipeadded=TRUE
+				newcase.state=3
```

Comment:
```suggestion
				var/obj/structure/girder/reactor/newcase= new /obj/structure/girder/reactor
				newcase.loc=src.loc
				newcase.pipeadded=TRUE
				newcase.state=3
				qdel(src)
```

Reply:
If you qdel(src) before referencing src.loc you're gonna send the new case to nullspace